### ISSUE TITLE
setup.cfg: UnicodeDecodeError: 'ascii' codec can't decode byte

### ISSUE THREAD

**Issue by ssbarnea** [pycontribs/jira#512]
It seem that `pbr` + `ChangeLog` inside `setup.cfg` do not make a hopeful marriage. 
```
py35-darwin develop-inst: /Users/ssbarnea/os/jira
ERROR: invocation failed (exit code 1), logfile: /Users/ssbarnea/os/jira/.tox/py35-darwin/log/py35-darwin-2.log
ERROR: actionid: py35-darwin
msg: developpkg
cmdargs: ['/Users/ssbarnea/os/jira/.tox/py35-darwin/bin/pip', 'install', '-e', '/Users/ssbarnea/os/jira[cli,opt,testing]']

Obtaining file:///Users/ssbarnea/os/jira
    Complete output from command python setup.py egg_info:
    ERROR:root:Error parsing
    Traceback (most recent call last):
      File "/Users/ssbarnea/os/jira/.tox/py35-darwin/lib/python3.5/site-packages/pbr/core.py", line 111, in pbr
        attrs = util.cfg_to_args(path, dist.script_args)
      File "/Users/ssbarnea/os/jira/.tox/py35-darwin/lib/python3.5/site-packages/pbr/util.py", line 251, in cfg_to_args
        kwargs = setup_cfg_to_setup_kwargs(config, script_args)
      File "/Users/ssbarnea/os/jira/.tox/py35-darwin/lib/python3.5/site-packages/pbr/util.py", line 315, in setup_cfg_to_setup_kwargs
        value += description_file.read().strip() + '\n\n'
      File "/Users/ssbarnea/os/jira/.tox/py35-darwin/lib/python3.5/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xe5 in position 691: ordinal not in range(128)
    error in setup command: Error parsing /Users/ssbarnea/os/jira/setup.cfg: UnicodeDecodeError: 'ascii' codec can't decode byte 0xe5 in position 691: ordinal not in range(128)

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /Users/ssbarnea/os/jira/
```

The problem does appear only on Python 3.x versions and after testing even the latest pbr==3.1.1 i was not able to find a clear way to avoid it.

So far, the only workaround is to disable the inclusion of the  autogenerated ChangeLog file from `setup.cfg` description.

**Comment by ssbarnea** [comment]
Fixed in 1.0.14, which was also released few moments ago.

**Comment by iksaif** [comment]
Are your sure this is resolved ? The Changelog is still in "description-file" and I get the following error:

```
15:18:23 Installing jiralerts script to build/bdist.linux-x86_64/dumb/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/bin
15:18:23 creating /home/jenkins/workspace/thirdparty-jiralerts-pre-submit/dist
15:18:23 Creating tar archive
15:18:23 removing 'build/bdist.linux-x86_64/dumb' (and everything under it)
15:18:23 bundle runtests: commands[2] | /tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/bin/python setup.py bdist_pex --bdist-dir=dist --pex-args=--disable-cache --no-wheel
15:18:24 running bdist_pex
15:18:27 **** Failed to install jira-1.0.14 (caused by: NonZeroExit("received exit code 1 during execution of `['/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/bin/python3', '-', 'bdist_egg', '--dist-dir=/tmp/tmpjuu3zaw3']` while trying to execute `['/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/bin/python3', '-', 'bdist_egg', '--dist-dir=/tmp/tmpjuu3zaw3']`",)
15:18:27 ):
15:18:27 stdout:
15:18:27 
15:18:27 Installed /tmp/tmpvs3aciu_/jira-1.0.14/.eggs/pytest_runner-4.0-py3.4.egg
15:18:27 
15:18:27 stderr:
15:18:27 zip_safe flag not set; analyzing archive contents...
15:18:27 ERROR:root:Error parsing
15:18:27 Traceback (most recent call last):
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pbr/core.py", line 111, in pbr
15:18:27     attrs = util.cfg_to_args(path, dist.script_args)
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pbr/util.py", line 249, in cfg_to_args
15:18:27     pbr.hooks.setup_hook(config)
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pbr/hooks/__init__.py", line 25, in setup_hook
15:18:27     metadata_config.run()
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pbr/hooks/base.py", line 27, in run
15:18:27     self.hook()
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pbr/hooks/metadata.py", line 26, in hook
15:18:27     self.config['name'], self.config.get('version', None))
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pbr/packaging.py", line 737, in get_version
15:18:27     version = _get_version_from_pkg_metadata(package_name)
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pbr/packaging.py", line 705, in _get_version_from_pkg_metadata
15:18:27     pkg_metadata = email.message_from_file(pkg_metadata_file)
15:18:27   File "/usr/lib64/python3.4/email/__init__.py", line 56, in message_from_file
15:18:27     return Parser(*args, **kws).parse(fp)
15:18:27   File "/usr/lib64/python3.4/email/parser.py", line 54, in parse
15:18:27     data = fp.read(8192)
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib64/python3.4/encodings/ascii.py", line 26, in decode
15:18:27     return codecs.ascii_decode(input, self.errors)[0]
15:18:27 UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5410: ordinal not in range(128)
15:18:27 error in setup command: Error parsing /tmp/tmpvs3aciu_/jira-1.0.14/setup.cfg: UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5410: ordinal not in range(128)
15:18:27 
15:18:27 
15:18:27 Traceback (most recent call last):
15:18:27   File "setup.py", line 36, in <module>
15:18:27     packages=setuptools.find_packages())
15:18:27   File "/usr/lib64/python3.4/distutils/core.py", line 148, in setup
15:18:27     dist.run_commands()
15:18:27   File "/usr/lib64/python3.4/distutils/dist.py", line 955, in run_commands
15:18:27     self.run_command(cmd)
15:18:27   File "/usr/lib64/python3.4/distutils/dist.py", line 974, in run_command
15:18:27     cmd_obj.run()
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pex/commands/bdist_pex.py", line 89, in run
15:18:27     pex_builder = build_pex(reqs, options, options_builder)
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pex/bin/pex.py", line 601, in build_pex
15:18:27     for dist in resolveds:
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pex/resolver.py", line 438, in resolve_multi
15:18:27     allow_prereleases):
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pex/resolver.py", line 376, in resolve
15:18:27     return resolver.resolve(resolvables_from_iterable(requirements, builder))
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pex/resolver.py", line 209, in resolve
15:18:27     dist = self.build(package, resolvable.options)
15:18:27   File "/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/lib/python3.4/site-packages/pex/resolver.py", line 177, in build
15:18:27     raise Untranslateable('Package %s is not translateable by %s' % (package, translator))
15:18:27 pex.resolver.Untranslateable: Package SourcePackage('https://pypi.python.org/packages/d9/bf/39343975b422f424513f7e83e6162f5826bd48e09263c6a42b2695d09a52/jira-1.0.14.tar.gz#md5=8f6d9776a4ff9e5d301aaacd8c27c71a') is not translateable by ChainedTranslator(EggTranslator, SourceTranslator)
15:18:27 ERROR: InvocationError: '/tmp/thirdparty-jiralerts-pre-submit11/com.github.criteo-forks.jiralerts/bundle/bin/python setup.py bdist_pex --bdist-dir=dist --pex-args=--disable-cache --no-wheel'
```

**Comment by ssbarnea** [comment]
Be sure you clone the repository again or read https://stackoverflow.com/questions/9813816/git-pull-after-forced-update in order to refresh it.

Mainly the fix involved rewriting bit of git history, hopefully something I will not have to do again.

**Comment by iksaif** [comment]
I'm only using things that are published on pypi (jira-1.0.14) and not the git repository itself.

**Comment by dbaxa** [comment]
@iksaif please ensure that your build machine sets `LANG` to a utf-8 friendly value, for example `en_US.UTF-8`

**Comment by iksaif** [comment]
That could be it, I'm not able to reproduce this with `LANG=C LC_ALL=C pip install jira -U --force-reinstall --no-cache` but I am with `LANG=C python setup.py bdist_pex --bdist-dir=dist --pex-args='--disable-cache --no-wheel'` Thanks for the hint !

**Comment by ssbarnea** [comment]
With this occasion I found about pex. Please continue to investigate the problem in order to find where is the bug. 

My goal is not enforce people to alter their locate in order to install jira library, this should work any/most locales.

**Comment by ssbarnea** [comment]
@iksaif Can please provide more info regarding on which platform you got the errors? I tried your line on tip of `master` branch on MacOS and worked perfectly.

**Comment by iksaif** [comment]
I can reproduce this using 'LANG=C tox' with tox calling setup.py. Since tox passes only a few variables, including LANG, that probably explain the error.

Any system with a correctly setup default UTF8 locale should work as expected.

**Comment by ssbarnea** [comment]
@iksaif Can you confirm this on current master? Please also specify your python platform, as you it may be key. If we confirm it, we should tune the tox.ini to catch it on CI.

**Comment by iksaif** [comment]
setup.py
```python
import setuptools

setuptools.setup(name='test',
      packages=[],
      install_requires=['jira'],
)
```

tox.ini
```ini
[testenv]
deps=pex
commands=python setup.py bdist_pex --bdist-dir=dist --pex-args='--disable-cache --no-wheel' --bdist-all
```

```bash
$ virtualenv venv -p python2
$ source venv/bin/activate
$ pip install tox
$ LANG=C tox
GLOB sdist-make: /tmp/setup.py
python inst-nodeps: /tmp/.tox/dist/test-0.0.0.zip
python installed: certifi==2018.1.18,chardet==3.0.4,defusedxml==0.5.0,idna==2.6,jira==1.0.14,oauthlib==2.0.7,pbr==4.0.0,pex==1.3.1,pkg-resources==0.0.0,requests==2.18.4,requests-oauthlib==0.8.0,requests-toolbelt==0.8.0,six==1.11.0,test==0.0.0,urllib3==1.22
python runtests: PYTHONHASHSEED='326123516'
python runtests: commands[0] | python setup.py bdist_pex --bdist-dir=dist --pex-args=--disable-cache --no-wheel --bdist-all
running bdist_pex
**** Failed to install jira-1.0.14 (caused by: NonZeroExit("received exit code 1 during execution of `['/tmp/.tox/python/bin/python3', '-', 'bdist_egg', '--dist-dir=/tmp/tmpj0byg8c2']` while trying to execute `['/tmp/.tox/python/bin/python3', '-', 'bdist_egg', '--dist-dir=/tmp/tmpj0byg8c2']`",)
):
stdout:

Installed /tmp/tmp5_u4rc27/jira-1.0.14/.eggs/pytest_runner-4.2-py3.6.egg

stderr:
zip_safe flag not set; analyzing archive contents...
ERROR:root:Error parsing
Traceback (most recent call last):
  File "/tmp/.tox/python/lib/python3.6/site-packages/pbr/core.py", line 96, in pbr
    attrs = util.cfg_to_args(path, dist.script_args)
  File "/tmp/.tox/python/lib/python3.6/site-packages/pbr/util.py", line 254, in cfg_to_args
    pbr.hooks.setup_hook(config)
  File "/tmp/.tox/python/lib/python3.6/site-packages/pbr/hooks/__init__.py", line 25, in setup_hook
    metadata_config.run()
  File "/tmp/.tox/python/lib/python3.6/site-packages/pbr/hooks/base.py", line 27, in run
    self.hook()
  File "/tmp/.tox/python/lib/python3.6/site-packages/pbr/hooks/metadata.py", line 26, in hook
    self.config['name'], self.config.get('version', None))
  File "/tmp/.tox/python/lib/python3.6/site-packages/pbr/packaging.py", line 794, in get_version
    version = _get_version_from_pkg_metadata(package_name)
  File "/tmp/.tox/python/lib/python3.6/site-packages/pbr/packaging.py", line 762, in _get_version_from_pkg_metadata
    pkg_metadata = email.message_from_file(pkg_metadata_file)
  File "/usr/lib/python3.6/email/__init__.py", line 54, in message_from_file
    return Parser(*args, **kws).parse(fp)
  File "/usr/lib/python3.6/email/parser.py", line 54, in parse
    data = fp.read(8192)
  File "/tmp/.tox/python/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5410: ordinal not in range(128)
error in setup command: Error parsing /tmp/tmp5_u4rc27/jira-1.0.14/setup.cfg: UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5410: ordinal not in range(128)


Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    install_requires=['jira'],
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/tmp/.tox/python/lib/python3.6/site-packages/pex/commands/bdist_pex.py", line 89, in run
    pex_builder = build_pex(reqs, options, options_builder)
  File "/tmp/.tox/python/lib/python3.6/site-packages/pex/bin/pex.py", line 601, in build_pex
    for dist in resolveds:
  File "/tmp/.tox/python/lib/python3.6/site-packages/pex/resolver.py", line 438, in resolve_multi
    allow_prereleases):
  File "/tmp/.tox/python/lib/python3.6/site-packages/pex/resolver.py", line 376, in resolve
    return resolver.resolve(resolvables_from_iterable(requirements, builder))
  File "/tmp/.tox/python/lib/python3.6/site-packages/pex/resolver.py", line 209, in resolve
    dist = self.build(package, resolvable.options)
  File "/tmp/.tox/python/lib/python3.6/site-packages/pex/resolver.py", line 177, in build
    raise Untranslateable('Package %s is not translateable by %s' % (package, translator))
pex.resolver.Untranslateable: Package SourcePackage('https://pypi.python.org/packages/d9/bf/39343975b422f424513f7e83e6162f5826bd48e09263c6a42b2695d09a52/jira-1.0.14.tar.gz#md5=8f6d9776a4ff9e5d301aaacd8c27c71a') is not translateable by ChainedTranslator(EggTranslator, SourceTranslator)
ERROR: InvocationError: '/tmp/.tox/python/bin/python setup.py bdist_pex --bdist-dir=dist --pex-args=--disable-cache --no-wheel --bdist-all'
____________________________________________________________________________________________________ summary _____________________________________________________________________________________________________
ERROR:   python: commands failed

```

**Comment by Masuzu** [comment]
@iksaif This seems to be solved in v1.0.15

**Comment by iksaif** [comment]
great ! will try it out. Thanks !
